test/services/BusinessServiceSpec.scala

```diff
@@ -165,19 +167,66 @@ class BusinessServiceSpec extends AnyWordSpecLike {
   }
 
   "getNetBusinessProfitOrLossValues" should {
-    "return NetBusinessProfitValues for a business" in {
-      val expectedResult = Right(aNetBusinessProfitValues)
-      val stubIFSBusinessDetailsConnector = StubIFSBusinessDetailsConnector(
+    "return NetBusinessProfitValues for a business" in new Test {
+      override def stubIFSBusinessDetailsConnector = StubIFSBusinessDetailsConnector(
         getBusinessIncomeSourcesSummaryResult = aBusinessIncomeSourcesSummaryResponse.asRight
       )
-      val stubIFSConnector = StubIFSConnector(
-        getPeriodicSummaryDetailResult = Future.successful(api1786DeductionsSuccessResponse.asRight),
+      override def stubIFSConnector = StubIFSConnector(
+        getPeriodicSummaryDetailResult =
+          Future.successful(api1786DeductionsSuccessResponse.copy(financials = FinancialsType(None, Some(IncomeTypeTestData.sample))).asRight),
         getAnnualSummariesResult = Right(api_1803.SuccessResponseSchema(None, None, None))
       )
-      val service = new BusinessServiceImpl(stubIFSBusinessDetailsConnector, StubMDTPConnector(), stubIFSConnector)
-      service.getNetBusinessProfitOrLossValues(journeyCtxWithNino).value.map { result =>
```

Comment:
This was a mistake I made previously and am now correcting. Using `.map { result => ` always passes without properly testing the assertions contained. There are some other instances of this that I will need to correct but I did some in the PR for now